astropy/utils/data_info.py

```diff
@@ -594,6 +605,136 @@ def getattrs(col):
 
         return out
 
+    @property
+    def format(self):
+        return self._attrs['format']
+
+    @format.setter
+    def format(self, format_string):
+
+        prev_format = self._attrs['format']
+
+        self._attrs['format'] = format_string  # set new format string
+
+        try:
+            # test whether it formats without error exemplarily
+            self.pformat(max_lines=1)
```

Comment:
Does this try/except work also for a `MaskedColumn` in which the first item is masked? I recall having to work around that case in the printing routines... It could either fail on the masked item, or fail to raise on something that would fail on a real number.

Reply:
Yes, this is a problem (though not one introduced by this PR).  But there are mitigating factors:
- The *actual* smallest allowed value of `max_lines` is 7.  This allows the internal code to be a bit simpler and not deal with the problems that would ensue from formatting only 1 line.  So for a typical table you actually get the first two and last two lines checked.
- In the event that all these are masked, the worst case is that later on one gets the same exception you would have seen when setting the `format`.

Reply:
.. and I found a bug, see #6812.  This should be a very quick review.